SyncConnectionStateErrorReset/SyncConnectionStateErrorReset/Models/Models.swift

```diff
@@ -0,0 +1,34 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import RealmSwift
+
+class Kiosk: Object, ObjectKeyIdentifiable {
+    @Persisted(primaryKey: true) var _id: ObjectId
+    @Persisted var name: String
+    @Persisted var userId: String
+    @Persisted var products: List<Product>
+}
+
+class Product: Object, ObjectKeyIdentifiable {
+    @Persisted(primaryKey: true) var _id: ObjectId
+    @Persisted var title: String
+    @Persisted var storeId: ObjectId
+    @Persisted var price: Double
+    @Persisted var numInStock: Int
+}
```

Comment:
Do you think `storeId` and `userId` should be indexed by the way?

Reply:
@elle-j  Even though indexes sped up queries using equality, they come with some tradeoffs, they take memory space in the realm file and slow some operations like inserts and update, I don't know if we should get this example too complicated with this, we have another examples/documentation when indexes are showcased.